### PR TITLE
youtube-dl: add example for the --format option

### PR DESCRIPTION
**Pull request by vastpeng**
- [x] The page is in the correct platform folder (`common/`, `linux/`, etc.)
- [x] The page follows the [content guidelines](/tldr-pages/tldr/blob/master/CONTRIBUTING.md#guidelines).

### CODE REVIEW

File: pages/common/youtube-dl.md [hunk 1]
```diff
@@ -6,6 +6,10 @@
 
 `youtube-dl {{https://www.youtube.com/watch?v=oHg5SJYRHA0}}`
 
+- Download the specific quality video or playlist:
```
Comment (waldyrious) [inline pages/common/youtube-dl.md]:
```suggestion
- Download a video or playlist at a specific quality:
```

(Although perhaps we should explicitly name this as "at the best available quality"? Which also leads me to wonder -- how is this particular example different from the default?)

**Comment by waldyrious** [comment]
@vastpeng please check the comment I left inline. Also, can you add one example for `--list-formats`, as @agnivade mentioned?

File: pages/common/youtube-dl.md [hunk 2]
```diff
@@ -6,6 +6,10 @@
...
+
+`youtube-dl -f {{bestvideo+bestaudio}} {{https://www.youtube.com/watch?v=oHg5SJYRHA0}}`
```
[the remark this answers is not in the document]
Comment (agnivade) [reply pages/common/youtube-dl.md]: This is not resolved.

**Comment by vastpeng** [comment]
Sure, I will do it later.

File: pages/common/youtube-dl.md [hunk 3]
```diff
@@ -6,6 +6,14 @@
 
 `youtube-dl {{https://www.youtube.com/watch?v=oHg5SJYRHA0}}`
 
+- List all formats from a video or playlist:
+
+`youtube-dl --list-formats {{https://www.youtube.com/watch?v=Mwa0_nE9H7A}}`
```
[the remark this answers is not in the document]
Comment (agnivade) [reply pages/common/youtube-dl.md]: I think the formats are video specific. And I would expect it to be that way. Is it confusing ?

File: pages/common/youtube-dl.md [hunk 4]
```diff
@@ -6,6 +6,14 @@
...
+
+- Download a video or playlist at a specific quality:
+
+`youtube-dl --format {{bestvideo+bestaudio}} {{https://www.youtube.com/watch?v=oHg5SJYRHA0}}`
```
[the remark this answers is not in the document]
Comment (agnivade) [reply pages/common/youtube-dl.md]: I think the current description is fine. It just means user is free to put whatever quality he/she wants. Changing it to best available quality would effectively mean to remove the tokens (`{{}}`), which should be there ideally.

File: pages/common/youtube-dl.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (waldyrious) [reply pages/common/youtube-dl.md]: Many commands (e.g. converters) have similar options to list available input/output formats. If we're sure in this case the output is specific to the video, we probably should make that a bit clearer in the description. For example:

`List all formats that a video or playlist is available in`

or

`List all output formats that can be used for a given a video or playlist`

(of course, the wording depends on what exactly the option is displaying, which I'm still a bit unsure about.)

File: pages/common/youtube-dl.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (waldyrious) [reply pages/common/youtube-dl.md]: > Changing it to best available quality would effectively mean to remove the tokens (`{{}}`), which should be there ideally.

So perhaps we should use a format string that actually does something different from the default, no?

[the remark this answers is not in the document]
Comment (vastpeng) [reply pages/common/youtube-dl.md]: I think the current description is fine.

From documentation:

`By default youtube-dl tries to download the best available quality.`

I have tried it, if the best available quality is only video format, there is no audio, the same as the best audio. But I want to download the best video with audio When I use YouTube-dl.

[the remark this answers is not in the document]
Comment (vastpeng) [reply pages/common/youtube-dl.md]: @waldyrious

You are right, the default is `bestaudio+bestvideo`. I just want to express that we can choose the format.

[the remark this answers is not in the document]
Comment (agnivade) [reply pages/common/youtube-dl.md]: @waldyrious - Makes sense to me. @vastpeng - Could you please change the token value to something other than `bestaudio+bestvideo` ? Thanks.

**Comment by agnivade** [comment]
@waldyrious - Are you okay with the new values ? Or you wanted some resolution specific values ?

**Comment by waldyrious** [comment]
> @waldyrious - Are you okay with the new values ? Or you wanted some resolution specific values ?

I don't think "worstvideo+worstaudio" is a good token value. I actually thought that @vastpeng was getting frustrated with our review comments and vented by picking those values, but after I went looking for the [youtube-dl documentation](https://github.com/rg3/youtube-dl/blob/master/README.md#format-selection), to my surprise I found that they're actually supported values for the `--format` option!

Still, I think we can pick a better example. In fact, the docs have a very nice [section](https://github.com/rg3/youtube-dl/blob/master/README.md#format-selection-examples) precisely with some useful examples. From those, perhaps this one can illustrate the usefulness of this option:

```sh
# Download best video-only format but no bigger than 50 MB
$ youtube-dl -f 'best[filesize<50M]'
```

What do you think?

**Comment by waldyrious** [comment]
By the way, in the docs linked above I found the following passage:

> By default youtube-dl tries to download the best available quality, i.e. if you want the best quality you **don't need** to pass any special options, youtube-dl will guess it for you by **default**.

...which confirms what we suspected above, and

> You can get the list of available format codes for particular video using `--list-formats`

...which also answers a question we had during review.

**Comment by agnivade** [comment]
Personally, I would want videos according to the height. So  `youtube-dl -f 'best[height<=480]'` is more useful to me.

**Comment by waldyrious** [comment]
Fine by me. Wanna make the call, or should we wait for someone else to comment?

**Comment by agnivade** [comment]
@mfrw / @pxgamer - Does that sound good ?

**Comment by vastpeng** [comment]
@agnivade 

I think any is OK, just to be explicit.

**Comment by vastpeng** [comment]
@waldyrious

Not, it’s beginning of my open source contribute. I can get a lot from it.

**Comment by mebeim** [comment]
I'm just going to leave this here: what about using something like `{{youtube_video_url}}` instead of *actual YouTube links*? I see two of them are good old rickrolls and one is a random Chinese video of a guy talking about aliens... I mean... wut?? :rofl:

**Comment by agnivade** [comment]
We usually prefer concrete examples in our pages. I think it is okay to have an actual youtube url. I didn't even bother to check the video You can suggest a better url if you want.

**Comment by mebeim** [comment]
Oh, yeah, I see, was just confused by the fact that it's mixed between real URLs and `{{url}}`. I would suggest to use the same `https://www.youtube.com/watch?v=oHg5SJYRHA0` anyway.

**Pull request merged**